tests/test_user.py

```diff
@@ -458,3 +461,446 @@ def test_user_settings_edit_dev_app(self, driver, session, fake):
+            pat_page.loading_indicator.here_then_gone()
+            pat_card = pat_page.get_pat_card_by_name(token_name)
+            assert pat_card
+            pat_link = pat_card.find_element_by_css_selector('a')
+
+            # Now click the PAT name link again to go back to the Edit PAT page
+            pat_link.click()
+            edit_page = user.EditPersonalAccessTokenPage(driver, verify=True)
+            assert edit_page.token_name_input.get_attribute('value') == token_name
+
+            # Click the Delete button again and this time click the Delete button on the
+            # Delete Token Modal
+            edit_page.scroll_into_view(edit_page.delete_button.element)
+            edit_page.delete_button.click()
+            delete_modal = edit_page.delete_pat_modal
+            assert delete_modal.token_name.text == token_name
+            delete_modal.delete_button.click()
+
+            # This time we should end up on the PAT list page
+            pat_page = user.PersonalAccessTokenPage(driver, verify=True)
+            pat_page.loading_indicator.here_then_gone()
+            pat_card = pat_page.get_pat_card_by_name(token_name)
+
+            # Verify that we don't find the PAT card this time since it was deleted
+            assert not pat_card
+        except Exception:
+            # As cleanup, delete the PAT using the api if the test failed for some
+            # reason and the PAT was not actually deleted.
+            pat_data = osf_api.get_user_pat_data(session, token_id=public_token_id)
+            if pat_data:
+                osf_api.delete_personal_access_token(session, token_id=public_token_id)
+
+    def test_user_settings_delete_PAT_from_list_page(
+        self, driver, session, fake, all_scopes
+    ):
+        """Delete a Personal Access Token from the User Settings Edit Personal Access
+        Token page in OSF. The test uses the OSF api to first create the personal access
+        token that will then be deleted using the Front End interface.
+        """
+        token_name = 'PAT created via api ' + fake.sentence(nb_words=1)
+        token_ids = osf_api.create_personal_access_token(
+            session,
+            name=token_name,
+            scopes='osf.users.profile_read osf.users.email_read',
+        )
+        public_token_id = token_ids[0]
+        private_token_id = token_ids[1]
+
+        try:
+            # Use the PAT so that it becomes registered in CAS
+            current_user = osf_api.current_user()
+            user_url = settings.API_DOMAIN + '/v2/users/{}/settings/emails/'.format(
+                current_user.id
+            )
+            headers = {'Authorization': 'Bearer ' + private_token_id}
+            r = requests.get(user_url, headers=headers)
+            # verify the node access response and some of the returned data
+            assert r.status_code == 200
+            assert r.json()['data'][0]['attributes']['confirmed']
```

Comment:
Ah, I like this one where we use a different endpoint to use the PAT (from the one used for the other delete tests.

Reply:
Thanks. Yeah this test uses different scopes for the PAT so may as well take advantage of the permissions that the scopes provide.